src/platform_impl/linux/wayland/window/mod.rs

```diff
@@ -457,6 +462,17 @@ impl Window {
         Ok(())
     }
 
+    pub fn request_user_attention(&self, request_type: Option<UserAttentionType>) {
+        if !self.windowing_features.xdg_activation() {
+            warn!("'request_user_attention' isn't supported.");
+            return;
+        }
+
+        let attention_request = WindowRequest::Attention(request_type);
+        self.window_requests.lock().unwrap().push(attention_request);
+        self.event_loop_awakener.ping();
```

Comment:
Just something I've noticed that could be considered for a future refactoring: In this file every single instance of `window_requests.lock().unwrap().push(_)` is followed up by `event_loop_awakener.ping()`. If those two must go together it would probably be a good idea to have a function or something that does this, rather than duplicating it everywhere.